crates/proto/src/rr/domain/name.rs

```diff
@@ -56,6 +56,35 @@ impl Name {
         Ok(())
     }
 
+    /// Randomize the case of ASCII alpha characters in a name
+    pub fn randomize_label_case(&mut self) {
+        use rand::distributions::{Distribution, Standard};
+        let mut rand = rand::thread_rng();
+
+        // 16 bits should be enough to cover *most* queries without making another call to the RNG.
+        // Setting this to u16 will cause the most significant bit to always be zero.
+        let mut rand_bits: i16 = 0;
+
+        for i in 0..self.label_data.len() {
```

Comment:
Suggest writing this as `for (i, &b) in self.label_data.iter().enumerate() { .. }`.

Reply:
I ended up with for (i, ref mut b) and eliminated all of the array indexes